### ISSUE TITLE
Generic/NestingLevel: improve error message

### ISSUE THREAD

**Issue by rodrigoprimo** [PHPCSStandards/PHP_CodeSniffer#763]
## Is your feature request related to a problem?

Similar to #743, both the warning and the error message displayed by the Generic.Metrics.NestingLevel sniff could be improved if they included the name of the function that caused the error.

Currently, the sniff displays the following messages:

```
  3 | WARNING | Function's nesting level (6) exceeds 5; consider refactoring the function (Generic.Metrics.NestingLevel.TooHigh)
 20 | ERROR   | Function's nesting level (11) exceeds allowed maximum of 10 (Generic.Metrics.NestingLevel.MaxExceeded)
```

I'm suggesting we update them to something like:

```
  3 | WARNING | Function nestingSix()'s nesting level (6) exceeds 5; consider refactoring the function (Generic.Metrics.NestingLevel.TooHigh)
 20 | ERROR   | Function nestingElevent()'s nesting level (11) exceeds allowed maximum of 10 (Generic.Metrics.NestingLevel.MaxExceeded)
```

- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [ ] I intend to create a pull request to implement this feature.

**Comment by jrfnl** [comment]
Valid suggestion, low priority.

Note: this would become more complex if #761 gets implemented as anonymous functions don't have a name.

**Comment by paulshryock** [comment]
Why is this needed if the line number is already provided?

What happens when someone writes a very long function name? If it's a class method, does this log the fully qualified class and method name? Does the error message end up spanning multiple lines in the terminal?

```bash
# do you really want to add this mess to the error message?

3 | WARNING | Function MyVendor\MyPackage\MyNamespace\MyClassName::myVeryLongMethodName()'s nesting level (6) exceeds 5; consider refactoring the function (Generic.Metrics.NestingLevel.TooHigh)
```

What happens when someone copy/pastes the error message into a search engine? Does the inclusion of their function name in the search query make it less likely that they will find the correct documentation they are looking for?

To me, this feels not worth the potential downsides.

I feel that the less "user input" which is logged to the terminal, the less risk there is.
